include/unifex/let_error.hpp

```diff
@@ -166,31 +171,47 @@ class _frcvr<Source, Func, Receiver, Error>::type {
 
   template(typename... Values)
     (requires receiver_of<Receiver, Values...>)
-  void set_value(Values&&... values) noexcept(
+  // Taking by value here to force a move/copy on the offchance the value
+  // objects live in the operation state, in which case destroying the
+  // predecessor operation state would invalidate it.
+  void set_value(Values... values) noexcept(
       is_nothrow_receiver_of_v<Receiver, Values...>) {
-    UNIFEX_ASSERT(op_ != nullptr);
-    unifex::set_value(std::move(op_->receiver_), (Values&&)values...);
+    // local copy, b/c cleanup deletes this
+    auto op = op_;
+    cleanup(op);
+    unifex::set_value(std::move(op->receiver_), std::move(values)...);
   }
 
   void set_done() noexcept {
-    UNIFEX_ASSERT(op_ != nullptr);
-    unifex::set_done(std::move(op_->receiver_));
+    // local copy, b/c cleanup deletes this
+    auto op = op_;
+    cleanup(op);
+    unifex::set_done(std::move(op->receiver_));
   }
 
   template(typename ErrorValue)
     (requires receiver<Receiver, ErrorValue>)
-  void set_error(ErrorValue&& error) noexcept {
-    UNIFEX_ASSERT(op_ != nullptr);
-    unifex::set_error(std::move(op_->receiver_), (ErrorValue&&)error);
+  // Taking by value here to force a copy on the offchance the error
+  // object lives in the operation state, in which
+  // case the call to cleanup() would invalidate them.
+  void set_error(ErrorValue error) noexcept {
+    // local copy, b/c cleanup deletes this
+    auto op = op_;
+    UNIFEX_TRY {
+      cleanup(op);
+      unifex::set_error(std::move(op->receiver_), std::move(error));
+    } UNIFEX_CATCH (...) {
+      unifex::set_error(std::move(op->receiver_), std::current_exception());
+    }
   }
 
 private:
-  void cleanup() noexcept {
+  void cleanup(operation* op) noexcept {
```

Comment:
This is `static`, right?
```suggestion
  static void cleanup(operation* op) noexcept {
```